packages/headless/src/features/analytics/analytics-utils.spec.ts

```diff
@@ -0,0 +1,31 @@
+import {createMockState} from '../../test';
+import {buildMockResult} from '../../test';
+import {partialDocumentInformation} from './analytics-utils';
+
+describe('analytics-utils', () => {
+  it('should extract documentation information with a single author', () => {
+    const result = buildMockResult();
+    result.raw['author'] = 'john';
+
+    expect(
+      partialDocumentInformation(result, createMockState()).documentAuthor
+    ).toBe('john');
+  });
+
+  it('should extract documentation information with multiple author', () => {
+    const result = buildMockResult();
+    result.raw['author'] = ['john', 'doe'];
+
+    expect(
+      partialDocumentInformation(result, createMockState()).documentAuthor
+    ).toBe('john;doe');
+  });
+
+  it('should extract document information when there is no author', () => {
+    const result = buildMockResult();
+    delete result.raw['author'];
```

Comment:
Just to make sure that this `delete` needed, is it that the author key must not exist? or should it also set the `documentAuthor` to `unknown` if it is present and set to e.g. `null` or an empty string if that is not possible.

Edit: I see the code checks for `null`/ `undefined` values, but it seems like an empty string would be sent as-is.